esmvalcore/preprocessor/_multimodel.py

```diff
@@ -451,3 +383,33 @@ def multi_model_statistics(products, span, statistics, output_products=None):
         products |= statistic_products
         return products
     return statistic_products
+
+
+def ensemble_statistics(products, output_products, statistics: list):
+    """Calculate ensemble statistics.
+
+    Parameters
+    ----------
+    statistics: list
+        List of operators to apply to the data ensemble
+    """
+    product_dict = defaultdict(set)
+    span = 'overlap'
+
+    for product in products:
+        dataset = '_'.join([product.attributes['project'],
+                            product.attributes['dataset'],
+                            ''.join(product.attributes['exp'])])  # TODO: clean this
+
+        product_dict[dataset].add(product)
+
+    for dataset, ensemble_products in product_dict.items():
+        statistic_products = multi_model_statistics(
+            ensemble_products,
+            span,
+            statistics,
+            output_products[dataset],
+        )
+        products |= statistic_products  # set union
```

Comment:
Maybe we should not add the ensemble statistics to the set of products, but replace the existing products altogether. Otherwise, if you apply multi-model statistics later, you will apply it to the combination of individual members and the ensemble mean.